backend/cfg/cfg.ml

```diff
@@ -424,3 +424,39 @@ let set_stack_offset (instr : _ instruction) stack_offset =
 
 let set_live (instr : _ instruction) live =
   if Reg.Set.equal instr.live live then instr else { instr with live }
+
+let terminator_is_goto (desc : terminator) =
+  match desc with
+  | Always _ -> true
+  | Raise _ | Tailcall _ | Call_no_return _ | Never | Parity_test _
+  | Truth_test _ | Float_test _ | Int_test _ | Switch _ | Return ->
+    false
+
+exception Malformed_cfg of string
+
+let can_raise_block block =
+  (* Recompute [can_raise] and check that instructions in the middle of the
+     block do not raise, i.e., only the terminator, or the last instruction
+     (when the terminator is just a goto) can raise. *)
+  let rec check = function
+    | [] -> false
+    | [last] ->
+      let res = can_raise_basic last.desc in
+      if res && not (terminator_is_goto block.terminator.desc)
```

Comment:
Why isn't is using `Cfg.can_raise_terminator`?

Reply:
Here we are just checking that if the last basic instruction of hte body can raise, then the terminator is a goto. 
`Cfg.can_raise_terminator` is used in L462.

Reply:
But isn't too restrictive? Why is a raising instruction followed
by _e.g._ a non-raising  integer test forbidden?

Reply:
It makes dataflow easier. That's what the constructions in linear_to_cfg and cfgize produces at the moment, but I have a patch to make all instructions that can raise into terminators, which would fix this odd constraint.

Reply:
Yes, it notice the code was just moved here. I thought
this pull request was the opportunity to dust various
bits, but it is OK to leave it as it is.